backend/src/test_calculator.py

```diff
@@ -10,6 +11,284 @@ def setUp(self):
     def test_add(self):
         self.assertEqual(self.calculator.addition(1, 2), 3)
 
+    def test_add_negative(self):
+        self.assertEqual(self.calculator.addition(-1, -2), -3)
+
+    def test_add_negative_and_positive(self):
+        self.assertEqual(self.calculator.addition(-7, 3), -4)
+
+    def test_add_float(self):
+        self.assertAlmostEqual(self.calculator.addition(0.5, 0.45), 0.95)
+
+    def test_add_float_negative(self):
+        self.assertAlmostEqual(self.calculator.addition(-0.5, -0.45), -0.95)
+
+    def test_add_str(self):
+        self.assertEqual(self.calculator.addition('ab', 'bc'), 'abbc')
+
+    def test_add_str_and_nub(self):
+        self.assertRaises(TypeError, self.calculator.addition, 'ab', 5)
+
+    def test_add_None(self):
+        self.assertRaises(TypeError, self.calculator.addition, None, None)
+
+    def test_add_arrays(self):
+        self.assertEqual(self.calculator.addition([1, 2], [3, 4]), [1, 2, 3, 4])
+
+    def test_add_nan(self):
+        self.assertTrue(math.isnan(self.calculator.addition(1, math.nan)))
+
+    def test_add_inf(self):
+        self.assertEqual(self.calculator.addition(1, math.inf), math.inf)
+
+    def test_add_complex(self):
+        self.assertEqual(self.calculator.addition(1, 1j), 1+1j)
+
+
+    def test_mul(self):
+        self.assertEqual(self.calculator.multiplication(2, 3), 6)
+
+    def test_mul_negative_and_positive(self):
+        self.assertEqual(self.calculator.multiplication(2, -4), -8)
+
+    def test_mul_double_negative(self):
+        self.assertEqual(self.calculator.multiplication(-2, -3), 6)
+
+    def test_mul_float(self):
+        self.assertEqual(self.calculator.multiplication(0.1, 0.5), 0.05)
```

Comment:
Кажется, что, при работе с числами с плавающей запятой, лучше всегда использовать assertAlmostEqual
```suggestion
        self.assertAlmostEqual(self.calculator.multiplication(0.1, 0.5), 0.05)
```